typing/typedecl.ml

```diff
@@ -1515,6 +1515,8 @@ let transl_with_constraint id ?fixed_row_path ~sig_env ~sig_decl ~outer_env
   && sdecl.ptype_private = Private then
     Location.deprecated loc "spurious use of private";
   let type_kind, type_unboxed_default =
+    (* Here, `man = None` indicates we have a "fake" module type built by
+       [Typetexp.create_package_mty] for a constraint on a package type. *)
```

Comment:
I would say that we have a fake "with constraint", rather than a fake "module type". (The with constraint is used in a weird way inside the typing of Ptyp_package, but the module type is in fact something real coming from the user.)

I went on to read the relevant code (thanks for pointing out the function name!) and realized that it was in fact dead : #11897. This means that we could probably simplify the code here, but I am not sure how, and I believe that we should keep it outside the scope of the current PR which is about the immediacy stuff.

Note: I'm not sure what to do of your (useful!) new comment if #11897 gets merged. The function and logic that it refers to will be gone. We could either say: "`man = None` used to indicate that ...", or just remove the comment altogether for now. In either case, I don't want to take more of your time on this unrelated tangent.

Reply:
(And as a heads up, I've rebased and force pushed to pick up #11898 and other changes in this branch, so you may be out of sync if you pull.)